src/osd/OpQueueItem.cc

```diff
@@ -31,7 +31,7 @@ void PGPeeringItem::run(
   PGRef& pg,
   ThreadPool::TPHandle &handle)
 {
-  osd->dequeue_peering_evt(pg.get(), evt, handle);
+  osd->dequeue_peering_evt(sdata, pg.get(), evt, handle);
```

Comment:
It's still needed for the !pg case, where we can't follow pg->shard->osdmap.

It could be replaced by an osdmap ref, but the shard pointer seemed like a more natural thing to pass to the dequeue method.

Reply:
Okay, makes sense.